buildSrc/src/main/kotlin/ProjectExtensions.kt

```diff
@@ -1,4 +1,12 @@
 import org.gradle.api.Project
+import org.gradle.kotlin.dsl.extra
+import org.gradle.kotlin.dsl.provideDelegate
 
 val Project.javaModuleName: String
     get() = "org." + this.name.replace('-', '.')
+
+val Project.versions: Versions
+    get() {
+        var versions: Versions? by rootProject.extra
+        return versions ?: Versions(rootProject).also { versions = it }
```

Comment:
Nice! I wasn't aware setting the delegate would store it in extra properties. Now that I think about it that makes perfect sense. 👍